contents/handbook/engineering/schema-changes.md

```diff
@@ -10,6 +10,16 @@ and require extra effort from the engineering team.
 
 Below are some important considerations to keep in mind regarding schema changes:
 
+## General considerations
+
+Before making a schema change, consider:
+- Do we need the schema change at all? Would this be better solved with an application-level code change instead?
+- Is my change backwards compatible? Both old and new code _will_ be running in parallel in both posthog cloud and self-hosted, so breaking changes can and will cause outages.
+- Can I deploy my schema change separately from application code change? For non-trivial changes, you want to deploy schema change first to ensure it's easy to roll back and if it's backwards compatible.
+- Am I doing a blocking migration? Migrations which lock huge tables can easily cause outages. 
+
+If you're doing anything tricky, it is worth reading up on zero-downtime migrations and make sure you know how the change will work operationally in practice.
```

Comment:
I couldn't find a great resource to link here in ~10 minutes. If you the reviewer have suggestions that would be :+1: